floto/specs/task/task.py

```diff
@@ -23,5 +38,26 @@ def _default_id(self, name, version, input):
         else:
             requires_string = ''
 
-        hash_ = hashlib.sha1((input_string+requires_string).encode()).hexdigest()[:10]
-        return '{}:{}:{}'.format(name, version, hash_)
+        hash_ = hashlib.sha1((input_string + requires_string).encode()).hexdigest()[:10]
+        return '{}:{}:{}:{}'.format(name, version, domain, hash_)
+
+    def serializable(self):
+        """Serializable representation of self
+
+        Returns
+        -------
+        dict
+        """
+        # remove None values
+        cpy = {k: v for k, v in self.__dict__.items() if v is not None}
+
+        # add type
+        module_name = '.'.join(self.__module__.split('.')[:-1])
+        cpy['type'] = module_name + '.' + self.__class__.__name__
+
+        # in the requires list, we need nothing but the task's id_
+        requires = cpy.get('requires')
+        if requires:
+            cpy['requires'] = [{'id_': t.id_} for t in requires]
```

Comment:
Hm, I agree that it's not beautiful (because so much depending on the json encoder, only half complete, a `serialize`would be much nicer etc), but this does the same as your former `remove_unnecessary_fields` in the json encoder and passes all tests. 
1. From the code there is no way to see what `requires` should be, it was inconsistently using strings and tasks, with a tendency towards tasks in the nearer pasts and examples. See TODO in line 21